### PR TITLE
qaoa heavy

### PR DESCRIPTION
**Pull request by soranjh**
This PR adds heavy benchmark functions for QAOA optimization.

### CODE REVIEW

**Review by mariaschuld — APPROVED** [review]
Nice, so we now have qaoa light/heavy and time/mem benchmarks...

File: benchmarks/asv/app_suite.py [hunk 1]
```diff
@@ -44,3 +43,29 @@ def time_minvertex(self, n_layers):
...
+        a graph."""
+        hyperparams = {'n_layers': n_layers}
+        benchmark_qaoa(hyperparams)
+
+class QAOA_heavy:
+    """Benchmark the QAOA algorithm for finding the minimum vertex cover of a large graph using
+    a large number of layers."""
+
+    n_layers = 5
+    graph = nx.complete_graph(20)
```
Comment (ixfoduap) [inline benchmarks/asv/app_suite.py:58]: @soranjh Could you please comment a bit more on how you arrived at these numbers? 20 qubits seems like quite a lot.

Also, why use the complete graph? It's an interesting choice. Min vertex cover is trivial there, it's just all the vertices. But then again the unconstrained cost Hamiltonian becomes more complicated

Comment (soranjh) [reply benchmarks/asv/app_suite.py:58]: 20 nodes is the largest number we can have without getting the benchmark timed out error. The time out default is currently 60 s. The combination of 5 layers / 20 nodes is the heaviest benchmark we can have within the 60 s limit. The choice of the complete graph is arbitrary and we can define any graph we want.

**Review by ixfoduap — APPROVED** [review]

**Review by albi3ro — COMMENTED** [review]
So on `asv run`, we can chose certain benchmarks by passing in a regular expression, like

```
asv run -bench app_suite
```

While in this current organization, we could select out heavy benchmarks by 
```
asv run -bench heavy
```

would there be an easy way to run non-heavy benchmarks? using `QAOA` would pull out both `QAOA` and `QAOA_heavy`. Maybe we should add `_light` to those tests, like `VQE_light`, so we could easily run

```
asv run -bench light
```

**Comment by soranjh** [comment]
> So on `asv run`, we can chose certain benchmarks by passing in a regular expression, like
> 
> ```
> asv run -bench app_suite
> ```
> 
> While in this current organization, we could select out heavy benchmarks by
> 
> ```
> asv run -bench heavy
> ```
> 
> would there be an easy way to run non-heavy benchmarks? using `QAOA` would pull out both `QAOA` and `QAOA_heavy`. Maybe we should add `_light` to those tests, like `VQE_light`, so we could easily run
> 
> ```
> asv run -bench light
> ```

Yes. I will add `_light` to those tests.

**Pull request merged**